docs/ota-client-guide/modules/ROOT/pages/virtual-secondaries.adoc

```diff
@@ -0,0 +1,45 @@
+= Use of Virtual Secondaries
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+
+Virtual secondaries can be used as a way to update a configuration file on the primary (where the primary is a standard OSTree image running an aktualizr). You can do this in two parts:
+
+* First you need a virtual secondary configured that can be used to update a configuration file.
+* Afterwards you can integrate it into your Yocto build.
+
+
+
+== Creating a Virtual Secondary
+
+. You can create a virtual secondary to run locally by following the steps on how to xref:simulate-device-basic.adoc[simulate a device without building a disk image].
++
+You can use the provided `.deb` releases mentioned and the `virtualsec.json` file, which you can edit to suit your needs: such as removing one of the two entries and setting the paths and names to whatever you prefer.
+. Afterwards, you will be able to download the arbitrary files to the location specified in the `firmware_path` variable.
+. Upload the files for the virtual secondary to the https://connect.ota.here.com[OTA Connect Portal].
+. Now go to the *Software versions* tab and click *Add Software*. You should make sure the *ECU Types* field matches the value specified in the `ecu_hardware_id` variable of the `virtualsec.json` file.
++
+NOTE: `ecu_hardware_id` must be unique per each file type that you'll like to deliver, otherwise an update of one of them will rewrite the others.
+
+
+== Updating via Virtual Secondaries
+
+To issue the update perform the following steps:
+
+. Go to your *Devices* tab and find your primary. (You can run `aktualizr-info` locally to get the name of the primary if you don't know it.)
+. Afterwards, select your virtual secondary and the file you uploaded to the *Software versions* tab.
++
+(For future reference, you can use the *Campaigns* tab to send updates to multiple devices simultaneously.)

```

Comment:
Should this "for future reference" line maybe get moved to after these steps? I think it might flow better to have the three steps together and then afterwards the separate line about campaigns, and then you won't need the parentheses, either.

Reply:
@patrickvacek in that case should I make it a third point to read as: 
"You can use the *Campaigns* tab to send updates to multiple devices simultaneously" ?

Reply:
I don't think it counts as a point in the series of steps here, so to me the idea of it being a sidenote (as it is now) is fine. I just think it reads better to have it at the end instead of in the middle of these steps.

Reply:
@patrickvacek I get your point now. I will amend it.